libs/common/src/vault/services/cipher.service.ts

```diff
@@ -904,8 +904,27 @@ export class CipherService implements CipherServiceAbstraction {
       });
     }
 
+    /*
+    Cozy customization
+
+    We do not want to clear the cache because reconstructing the cache involves decrypting all ciphers
+    which is very costly with our papers and contacts. Instead we manually update decrypted ciphers below.
+
     await this.clearCache();
+    */
     await this.stateService.setEncryptedCiphers(ciphers);
+
+    // Cozy customization
+    const decryptedCiphers = await this.stateService.getDecryptedCiphers();
+    if (decryptedCiphers == null) {
+      return;
+    }
+
+    const ids = typeof id === "string" ? [id] : id;
+    const removed = decryptedCiphers.filter((c) => !ids.includes(c.id));
+
+    await this.stateService.setDecryptedCiphers(removed);
+    // Cozy customization end
```

Comment:
nit: use the following comment form to ease enabling/disabling the entire block and to be homogenous with other customizations

```suggestion
    // Cozy customization
    // We do not want to clear the cache because reconstructing the cache involves decrypting all ciphers
    // which is very costly with our papers and contacts. Instead we manually update decrypted ciphers below.
    /*
    await this.clearCache();
    /*/
    const decryptedCiphers = await this.stateService.getDecryptedCiphers();
    if (decryptedCiphers == null) {
      return;
    }

    const ids = typeof id === "string" ? [id] : id;
    const removed = decryptedCiphers.filter((c) => !ids.includes(c.id));

    await this.stateService.setDecryptedCiphers(removed);
    //*/
    await this.stateService.setEncryptedCiphers(ciphers);
```